src/display/views/threads/components/CheckboxTable.js

```diff
@@ -54,6 +55,37 @@ class CheckboxTable extends React.Component {
 		return {};
 	}
 
+	getTdProps(state, rowInfo, column, instance) {
```

Comment:
Logic looks good here, but can we possibly consolidate this into the existing `_getTdProps` file that you are already modifying below?

Reply:
Is the intent to avoid the boolean return from the click handler? (This didn't seem great to me either.)

I can explore this, but if I'm understanding correctly, this would mean passing in a callback to perform the actual toggle from that standalone function, meaning _getTdProps ends up calling setState, which is also a bit odd.

Reply:
I don't necessarily mean refactoring the logic you're making use of - more that since we have a `_getTdProps` file, perhaps this function should also live there (maybe as a separate export), just to keep related code together as much as possible. 

(It was sort of an off-the-cuff suggestion, though, so if there's a solid reason not to, I'm not super stressed about it. :D )

Reply:
I tried forklifting this out, but the parameter passing got to be a bit much. I explored what I first thought you meant instead, in the most recent commit, and I think the result is an improvement. Most of the weird code here was for proxying the click, which is now gone. We do still need to proxy the props factory though.